knox/knox.sh

```diff
@@ -24,6 +24,37 @@ readonly KNOX_GW_CONFIG="$(sudo -u knox mktemp -d -t knox-init-action-config-XXX
 
 readonly KNOX_HOME=/usr/lib/knox
 
+# Detect dataproc image version from its various names
+if (! test -n "${DATAPROC_IMAGE_VERSION}") && test -n "${DATAPROC_VERSION}"; then
+  DATAPROC_IMAGE_VERSION="${DATAPROC_VERSION}"
+  echo $DATAPROC_IMAGE_VERSION
+fi
+
+get_dataproc_sub_minor_version() {
+  # Setting Cluster name and region
+  hostname_output=$(hostname -A | tr -d '[:space:]' | awk -F "." '{print $1}')
+  if [[ "$hostname_output" == *"-m" ]]; then
+    CLUSTER="${hostname_output%-*}"
+  else
+    CLUSTER=$(hostname -A  |tr -d '[:space:]' | awk -F "." '{print $1}' | sed 's/\(.*\)-[^-]*-\([^-]*\)$/\1/')
+  fi
+  ZONE_PATH=$(curl "http://metadata.google.internal/computeMetadata/v1/instance/zone" -H "Metadata-Flavor: Google")
+  ZONE=$(echo $ZONE_PATH |awk -F "/" '{print $4}')
+  REGION="${ZONE%-*}"
+
+
+  # Get Subminor version of Dataproc
+  SUB_MINOR_VERSION=$(gcloud dataproc clusters describe "$CLUSTER" --region "$REGION" | grep "imageVersion" | awk -F "." '{print $3}' | sed 's/-.*//')
+  echo "$SUB_MINOR_VERSION"
+}
+
+
+# Function to perform common operations in case of specific version_sub_minor version
+apply_changes() {
```

Comment:
Please use a more descriptive function name - comment_repo_line would be more reasonable.